From: wuyifeng <wuyifeng@nd.com.cn>
Date: Wed, 12 Jul 2023 16:59:24 +0800
Subject: AP9-17089 As a guest user, I would like to see the copyright and licensing information for Chromium.

Fix version: 2.0.0
Add 'Copyright'
Hide 'Google Terms of Service',  'ChromeOS Additional Terms of Service', and 'Google Privacy Policy'.
Show 'Menu -> Settings' in Guest mode.
Hide all items of Settings page except 'About Chrome' in Guest mode.

---
 .../res/xml/legal_information_preferences.xml | 13 +++--
 .../AppMenuPropertiesDelegateImpl.java        |  1 -
 .../chrome/browser/settings/MainSettings.java | 47 +++++++++++++++++++
 3 files changed, 57 insertions(+), 4 deletions(-)

diff --git a/chrome/android/java/res/xml/legal_information_preferences.xml b/chrome/android/java/res/xml/legal_information_preferences.xml
--- a/chrome/android/java/res/xml/legal_information_preferences.xml
+++ b/chrome/android/java/res/xml/legal_information_preferences.xml
@@ -8,6 +8,10 @@ found in the LICENSE file.
 <PreferenceScreen
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto">
+    <org.chromium.chrome.browser.about_settings.HyperlinkPreference
+        android:key="copyright"
+        android:title="@string/copyright"
+        app:url="@string/copyright_url" />
     <org.chromium.chrome.browser.about_settings.HyperlinkPreference
         android:key="open_source_license"
         android:title="@string/open_source_license_title"
@@ -15,13 +19,16 @@ found in the LICENSE file.
     <org.chromium.chrome.browser.about_settings.HyperlinkPreference
         android:key="google_terms_of_service"
         android:title="@string/google_terms_of_service_title"
-        app:url="@string/google_terms_of_service_url" />
+        app:url="@string/google_terms_of_service_url"
+        app:isPreferenceVisible="false" />
     <org.chromium.chrome.browser.about_settings.HyperlinkPreference
         android:key="chrome_additional_terms_of_service"
         android:title="@string/chrome_additional_terms_of_service_title"
-        app:url="@string/chrome_additional_terms_of_service_url" />
+        app:url="@string/chrome_additional_terms_of_service_url"
+        app:isPreferenceVisible="false" />
     <org.chromium.chrome.browser.about_settings.HyperlinkPreference
         android:key="privacy_policy"
         android:title="@string/privacy_policy_title"
-        app:url="@string/google_privacy_policy_url" />
+        app:url="@string/google_privacy_policy_url"
+        app:isPreferenceVisible="false" />
 </PreferenceScreen>
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java b/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
@@ -559,7 +559,6 @@ public class AppMenuPropertiesDelegateImpl implements AppMenuPropertiesDelegate
         final boolean isLogin = PrometheanManager.INSTANCE.isLogin();
         menu.findItem(R.id.all_bookmarks_menu_id).setVisible(isLogin);
         menu.findItem(R.id.bookmark_all_tabs_menu_id).setVisible(isLogin);
-        menu.findItem(R.id.preferences_id).setVisible(isLogin);
         menu.findItem(R.id.view_source_id).setVisible(false);
         menu.findItem(R.id.exit_id).setVisible(false);
 
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
@@ -25,6 +25,7 @@ import androidx.preference.Preference;
 import androidx.preference.PreferenceFragmentCompat;
 
 import org.chromium.base.ContextUtils;
+import org.chromium.base.PrometheanManager;
 import org.chromium.base.supplier.ObservableSupplier;
 import org.chromium.chrome.R;
 import org.chromium.chrome.browser.flags.ChromeFeatureList;
@@ -74,6 +75,16 @@ public class MainSettings extends PreferenceFragmentCompat
     public static final String PREF_NOTIFICATIONS = "notifications";
     public static final String PREF_DOWNLOADS = "downloads";
     public static final String PREF_DEVELOPER = "developer";
+    public static final String PREF_BASICS_SECTION = "basics_section";
+    public static final String PREF_AUTOFILL_PAYMENT_METHODS = "autofill_payment_methods";
+    public static final String PREF_AUTOFILL_ADDRESSES = "autofill_addresses";
+    public static final String PREF_ADBLOCK = "adblock";
+    public static final String PREF_ADVANCED_SECTION = "advanced_section";
+    public static final String PREF_ACCESSIBILITY = "accessibility";
+    public static final String PREF_CONTENT_SETTINGS = "content_settings";
+    public static final String PREF_USERAGENT_SETTINGS = "useragent_settings";
+    public static final String PREF_USERSCRIPTS_SETTINGS = "userscripts_settings";
+    public static final String PREF_LANGUAGES = "languages";
 
     // Used for elevating the privacy section behind the flag (see crbug.com/1099233).
     public static final int PRIVACY_ORDER_DEFAULT = 18;
@@ -122,6 +133,42 @@ public class MainSettings extends PreferenceFragmentCompat
     public void onResume() {
         super.onResume();
         updatePreferences();
+        hideAllExceptAboutChromeWhenGuest();
+    }
+
+    private void hideAllExceptAboutChromeWhenGuest() {
+        final boolean isLogin = PrometheanManager.INSTANCE.isLogin();
+        if (isLogin) {
+            return;
+        }
+        String[] prefs = {
+                PREF_ACCOUNT_AND_GOOGLE_SERVICES_SECTION,
+                PREF_BASICS_SECTION,
+                PREF_SEARCH_ENGINE,
+                PREF_PASSWORDS,
+                PREF_AUTOFILL_PAYMENT_METHODS,
+                PREF_AUTOFILL_ADDRESSES,
+                PREF_PRIVACY,
+                PREF_ADBLOCK,
+                PREF_NOTIFICATIONS,
+                PREF_UI_THEME,
+                PREF_ADVANCED_SECTION,
+                PREF_HOMEPAGE,
+                PREF_TOOLBAR_SHORTCUT,
+                PREF_ACCESSIBILITY,
+                PREF_CONTENT_SETTINGS,
+                PREF_USERAGENT_SETTINGS,
+                PREF_USERSCRIPTS_SETTINGS,
+                PREF_LANGUAGES,
+                PREF_DOWNLOADS,
+                PREF_DEVELOPER
+        };
+        for (String pref : prefs) {
+            Preference p = findPreference(pref);
+            if (p != null) {
+                p.setVisible(false);
+            }
+        }
     }
 
     private void createPreferences() {
--

